esphome/components/xiaomi_ble/xiaomi_ble.cpp

```diff
@@ -377,7 +384,9 @@ bool report_xiaomi_results(const optional<XiaomiParseResult> &result, const std:
   if (result->button_press.has_value()) {
     ESP_LOGD(TAG, "  Button: %s", (*result->button_press) ? "pressed" : "");
   }
-
+  if (result->has_smoke.has_value()) {
+    ESP_LOGI(TAG, "  Smoke: %s", (*result->has_smoke) ? "yes" : "no");
```

Comment:
```suggestion
    ESP_LOGI(TAG, "  Smoke: %s", YESNO(*result->has_smoke));
```

Be aware that it prints in upper case.